src/renderer_vk.cpp

```diff
@@ -5702,13 +5757,22 @@ VK_DESTROY
 		uint32_t statsKeyType[2] = {};
 
 		VkSemaphore renderWait = m_presentDone[m_backBufferColorIdx];
-		VK_CHECK(vkAcquireNextImageKHR(m_device
+		VkResult result = vkAcquireNextImageKHR(m_device
 				, m_swapchain
 				, UINT64_MAX
 				, renderWait
 				, VK_NULL_HANDLE
 				, &m_backBufferColorIdx
-				) );
+				);
+		if (VK_ERROR_OUT_OF_DATE_KHR == result)
+		{
+			m_needToRefreshSwapchain = true;
+			return;
+		}
+		else
+		{
+			VK_CHECK(result);
```

Comment:
Same as for `vkQueuePresentKHR`; should gracefully handle `VK_SUBOPTIMAL_KHR` being returned here.